### ISSUE TITLE
Possibility to theme toast component

### ISSUE THREAD

**Issue by tomdohnal** [chakra-ui/chakra-ui#2736]
# 🚀 Feature request

I'd like to have the possibility to theme the component that gets rendered after calling
```javascript
const addToast = useToast()

addToast(...)
```

## 🧱 Problem Statement / Justification

Theming components is a straightforward way to make the ChakraUI components look compliant with a brand design.

## ✅ Proposed solution or API

Just like I define a `Button` component theme, I'd like to have the possibility to do the same for the `Toast` component.

## ↩️ Alternatives

An alternative is to use a `render` function as an argument when calling `addToast(...)` or provide a custom toast component to the `ToastProvider`

## 📝 Additional Information

Additionally, when looking at the source code, I noticed that the toast component renders a `Alert` from React UI. I wonder if `Alert` component from ChakraUI itself couldn't be used rather. (It might decrease the bundle size)

**Comment by segunadebayo** [comment]
Hi @tomdohnal,

Thanks for reporting this. Toasts can look differently depending on the use-case. I'd recommend creating your own Toast component that's connected to the theme and style/design it based on your project. Then pass that to the `render` function.

You can follow this guide: https://chakra-ui.com/docs/theming/component-style to learn how to create yours.

Feel free to hop on Discord if you need any support with this.

Cheers.

PS: We're looking to update the toast component code pretty soon so yeah, we will add support for passing a custom toast component to a `ToastProvider`.

**Comment by kb-ally** [comment]
Looks there was parent div 
`<div class="chakra-toast__inner" data-reach-alert="true" style="pointer-events: auto;max-width: 560px;min-width: 300px;margin: 0.5rem;">`

which adds max-width: 560px, causes this behaviour 

overriding max-width of `chakra-toast__inner` can solve. Is there any better ways to solve it ?

**Comment by gj1118** [comment]
@segunadebayo I am trying to use something like this 

First , I import my theme
```import theme from '../../theme/theme';```

Note that my custom theme as `brand` colours defined in the extend theme

Then I try to use `createStandaloneToast` to let the toast use my theme , like so 

```const toast = createStandaloneToast({theme: theme})```

but apparently this is not working. Can you please let me know what else I should be doing in order to make the toastr use my theme please ?

Thanks

**Comment by Hexiro** [comment]
> @segunadebayo I am trying to use something like this
> 
> First , I import my theme `import theme from '../../theme/theme';`
> 
> Note that my custom theme as `brand` colours defined in the extend theme
> 
> In my theme file, I have extended it as follows
> 
> ```
>     toast: {
>         variants: {
>           information: props => ({
>             color: mode( 'gray.800','white')(props),
>             backgroundColor: "red",
>           }),
>         },
>       },
> ```
> 
> Then I try to use `createStandaloneToast` to let the toast use my theme , like so
> 
> `const toast = createStandaloneToast({theme: theme})`
> 
> and I am using the above created variant, like so
> 
> ```
>     toast({
>       title: 'Nothing to cancel',
>       position:"top",
>       variant:"information",
>       description:"You can't cancel your order if you haven't placed it yet",
>       status: 'error',
>       isclosable: true,
>     })
> ```
> 
> but apparently this is not working. Can you please let me know what else I should be doing in order to make the toastr use my theme please ?
> 
> Thanks

Hello @gj1118, if you're still looking for answers to this, this stack overflow answer was able to help me:
[https://stackoverflow.com/a/70378452/10830115](https://stackoverflow.com/a/70378452/10830115)

TL;DR: The [Toast](https://chakra-ui.com/docs/feedback/toast) component uses [Alert](https://chakra-ui.com/docs/feedback/alert) under the hood, so you have to override Alert's styles
